File: src/cs/matrix_result.cs
public class MatrixResult
{
    public MatrixResult(int[,] matrix, int sum, int side, bool success)
    {
        Matrix = matrix;
        Sum = sum;
        Side = side;
        Success = success;
    }

    public int[,] Matrix { get; }
    public int Sum { get; }
    public int Side { get; }
    public bool Success { get; }
}

File: src/cpp/matrix_result.cpp
#pragma once
#include <vector>

using std::vector;

class MatrixResult{
    public:
    vector<vector<int>> Matrix;
    int Sum;
    int Side;
    bool Success;

    MatrixResult(vector<vector<int>> matrix, int sum, int side, bool success){
        Matrix = matrix;
        Sum = sum;
        Side = side;
        Success = success;
    }
};